### ISSUE TITLE
BigInteger as double causes System.InvalidCastException: Object must implement IConvertible 

### ISSUE THREAD

**Issue by petsuter** [JamesNK/Newtonsoft.Json#2375]
### Source/destination types

```csharp
BigInteger / double
```

### Source JSON

```javascript
100000000000000000000000000000000000
```

### Expected behavior

Parse as `1e35`

### Actual behavior

System.InvalidCastException: Object must implement IConvertible.

### Steps to reproduce

```csharp
Console.WriteLine("{0}", Newtonsoft.Json.Linq.JToken.Parse("1e35").Value<double>()); // works
Console.WriteLine("{0}", Newtonsoft.Json.Linq.JToken.Parse("100000000000000000000000000000000000").Value<double>()); // throws
```

**Comment by elgonzo** [comment]
NOT confirmed!

Note that there is no parameterless `Jtoken.Value<T>()` method in the current version of Json.NET. There is only a ``Jtoken.Value<T>(object key)` method, so your example code does not even compile/work with the current library.

Try the following instead:

`
var j = Newtonsoft.Json.Linq.JToken.Parse("100000000000000000000000000000000000");
Console.WriteLine(j.ToObject<double>());
`

**Comment by elgonzo** [comment]
It seems it is specifically the implementation of the parameterless `Value<T>()` extension method that suffers from this problem.

For example, the `ToObject<T>()` method seems to work just fine:
```
Console.WriteLine("{0}", Newtonsoft.Json.Linq.JToken.Parse("100000000000000000000000000000000000").ToObject<double>());
```

**Comment by petsuter** [comment]
Is there a downside to using `ToObject<double>()` in the meantime?

What is the preferred way in general (also e.g. between `jObject["key"].Value<string>()` vs. `jObject.Value<string>("key")`)?

**Comment by petsuter** [comment]
> it is specifically the implementation of the "parameter-less" `Value<T>()` extension method that suffers from this problem

The method `Value<T>(key)` with a key parameter has the same problem:

```
Console.WriteLine("{0}", Newtonsoft.Json.Linq.JObject.Parse("{'key': 100000000000000000000000000000000000}").Value<double>("key"));
```

**Comment by elgonzo** [comment]
@petsuter
you are right. I tried to remember why i said that, but i don't recall anymore :-(
